### ISSUE TITLE
Opening glob pattern over scp

### ISSUE THREAD

**Issue by tubbsd** [vim/vim#3408]
I recently attempted opening files like vim -o scp://hostname/*.cpp

While it did not open several files, it did open the last file in ls-order.  I edited this file and saved it, but it was saved as, literally, *.cpp on the remote host.

Interesting behavior but likely not desirable.

**Comment by chrisbra** [comment]
This cannot possibly work. The `*` will be expanded by your shell and Vim does not even see it (unless there is no .cpp file in your current directory, in which case Vim will see the `*` name. Also, if there are several files matching, this will call `vim -o scp://hostname/foobar.cpp foobar1.cpp` so the rest of the files would be taken from the current directory.

**Comment by vim-ml** [comment]
On 2018-09-07, Christian Brabandt wrote:
> This cannot possibly work. The * will be expanded by your shell and Vim does
> not even see it (unless there is no .cpp file in your current directory, in
> which case Vim will see the * name. Also, if there are several files matching,
> this will call vim -o scp://hostname/foobar.cpp foobar1.cpp so the rest of the
> files would be taken from the current directory.

I don't understand why you say that this cannot possibly work.  It
seems to me to be just a limitation of netrw that scp works with
only one file.

I just tried the following command:

    scp hostname:*.txt .

and it successfully copied all the *.txt files in my home directory
on hostname to the current directory on the local machine.  If netrw
handles a command such as the OP's

    vim -o scp://hostname/*.cpp 

differently, that seems like a limitation of netrw--albeit perhaps
a reasonable one--rather than something that cannot work.

Regards,
Gary

**Comment by tubbsd** [comment]
It is 100% possible and does what I have said.  1) the glob will NOT be
expanded by the shell locally as there is an scp pathname in front of the
glob. 2) I had no cpp files locally when I did this, so nothing to expand
and 3) a user could prevent the expansion (but I didn't) with a backslash
or quoting.  It behaves exactly as I described and I've verified it
multiple times now.  :f on the session says scp://hostname/*.cpp but the
file it actually opened on the remote end is main.cpp.  If I :w the file
it'll save as *.cpp on the remote end.
